knative-operator/cmd/manager/main.go

```diff
@@ -161,26 +133,27 @@ func main() {
 	}
 }
 
-// serveCRMetrics gets the Operator/CustomResource GVKs and generates metrics based on those types.
-// It serves those metrics on "http://metricsHost:operatorMetricsPort".
-func serveCRMetrics(cfg *rest.Config) error {
-	// Below function returns filtered operator/CustomResource specific GVKs.
-	// For more control override the below GVK list with your own custom logic.
-	filteredGVK, err := k8sutil.GetGVKsFromAddToScheme(apis.AddToScheme)
+func setupMonitoring(ctx context.Context, cfg *rest.Config) error {
+	config, err := rest.InClusterConfig()
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to create cluster config", err)
 	}
-	// Get the namespace the operator is currently deployed in.
-	operatorNs, err := k8sutil.GetOperatorNamespace()
+	cl, err := client.New(config, client.Options{})
```

Comment:
Need to do this since manager's client wont work.

Reply:
Let me double check because for some reason it didnt bring back the namespace object.

Reply:
The client coming from the manager is a [split client](https://sdk.operatorframework.io/docs/golang/references/client/#default-client) so it reads from cache, the error I get is:

```
{"level":"error","ts":"2020-07-30T11:20:54.750Z","logger":"cmd","msg":"Failed to start monitoring","error":"failed to setup monitoring resources: Namespace \"openshift-serverless\" not found","stacktrace":"github.com/openshift-knative/serverless-operator/knative-operator/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/src/github.com/openshift-knative/serverless-operator/knative-operator/vendor/github.com/go-logr/zapr/zapr.go:128\nmain.main\n\t/go/src/github.com/openshift-knative/serverless-operator/knative-operator/cmd/manager/main.go:122\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:203"}
```
I will try check if cache sync works here, because I suspect cache prohibits getting the resource, although it shouldnt.

Reply:
@markusthoemmes  Ok now I recall at this point cache has not been started so there is no way to make it work. Usually clients are used in reconciliation funcs where things are setup for you. Since mgr.Start() is blocking I cannot use this client after.